integration_tests/tests/match/match.cy.ts

```diff
@@ -145,6 +145,44 @@ context('Placement Requests', () => {
     occupancyViewPage.shouldShowOccupancyCalendar(premiseCapacity)
   })
 
+  it('allows me to submit invalid dates in the book your placement form on occupancy view page and displays appropriate validation messages', () => {
+    const premisesName = 'Hope House'
+    const apType = 'normal'
+    const durationDays = 15
+    const startDate = '2024-07-23'
+    const endDate = '2024-08-07'
+    const totalCapacity = 10
+
+    // Given I am signed in as a cru_member
+    signIn(['cru_member'], ['cas1_space_booking_create'])
+
+    // And there is a placement request waiting for me to match
+    const person = personFactory.build()
+    const placementRequest = placementRequestDetailFactory.build({ person })
+    const premiseCapacity = cas1PremiseCapacityFactory.build({ premise: { bedCount: totalCapacity } })
+
+    cy.task('stubPlacementRequest', placementRequest)
+    cy.task('stubPremiseCapacity', { premisesId: premiseCapacity.premise.id, startDate, endDate, premiseCapacity })
+
+    // When I visit the occupancy view page
+    const occupancyViewPage = OccupancyViewPage.visit(
+      placementRequest,
+      startDate,
+      durationDays,
+      premisesName,
+      premiseCapacity.premise.id,
+      apType,
+    )
+
+    // And I submit invalid dates in the "Book your placement" form
+    occupancyViewPage.shouldShowMatchingDetails(totalCapacity, startDate, durationDays, placementRequest)
+    occupancyViewPage.shouldFillBookYourPlacementFormDates('2024-11-25', '2024-11-24')
+    occupancyViewPage.clickContinue()
+
+    // Then it should show validation messages
+    occupancyViewPage.shouldShowErrorSummaryAndErrorMessage('The departure date must be after the arrival date')
+  })
+
```

Comment:
I'm not sure there is a test that covers the happy path.  The next test books a space, but starts on the booking page, rather than the occupancy view.

Reply:
okay, no worries. So just so I understand the e2e need to cover the happy paths. And then integration tests must handle happy path as well as "un-happy" paths? So we always right both integration/e2e then I assume and integration must cover all scenarios?

Just trying to get a handle on the testing standards - thanks :)

Reply:
p.s. ` next test books a space` I assume you mean the `allows me to book a space` this pre-existed the occupancy view page so have not touched this one. I could refactor that test to start at occupancy view but if this is needed would it not have previously started at the search page (which previously redirected to the confirmation page)

Happy to do whatever is needed, just want to make sure we're headed in the right direction.

Thanks again!

Reply:
TBH, I'm not quite sure where the testing standards lie either as we seem to have a lot of tests yet a lot of holes.

The limitation with the E2E tests is that the CI pipeline only runs them against `dev` _after_ the branch is merged to `main` so we really need coverage before that point as it's a real pain winding back from any breakages that get through. The integration tests are really just browser tests as the whole API is mocked but at least they allow us to do short journeys beyond what unit tests can do. The unit tests don't even render the templates so there's a lot of things they won't pick up.

From what I can see, the `allows me to book a space` integration test creates the space starting on the confirmation page, but we don't have a test that goes from the occupancy view page, populated with valid dates, and checks that submission redirects to the confirmation page with the correct dates populated.

I hope that helps.

Reply:
okey doke. I understand now, so we are aiming for full coverage with integration tests so we are covered before we merge PR to protect us against broken pipelines etc. And then happy path only for e2e tests as this is interacting with the `dev` env for real (giving us confidence in the upstream service integrations).

I will align with the above. Thanks for explaining!

I was previously avoiding doing it in both places to reduce build time and more generally avoid duplicating effort across `Playright` and `Cyprus` but tbf the build time is fairly inconsequential and doesn't hurt to have it in both places.

I guess with the recent emergence of the `npm run test:e2e:local-dev-upstream:ui` (working alongside `ap-tools server start --local-ui --local-api-dev-upstream`) we could consider re-evaluating the balance across `Playright` / `Cyprus` in terms of coverage. I say this as this enables us to simulate locally what the the pipeline does and therefore have more confidence we won't break the build on merge. I realise however this is a new thing and guess overall perhaps it pays to have as much coverage as possible anyway. Food for thought and one for another time :)

Anyways, will crack on with upping the integration test coverage and thanks for explaining the rationale / standards so I know what to aim for going forward.

Thanks again!